File: log-analyzer.py
import re
import json
import sys

# Read data from stdin
data = sys.stdin.read()

# Define the regex pattern
pattern = r"Benchmark (\d+)[^]*?-t (\d+)[^]*?Task times: LongSummaryStatistics\{(.+)\}[^§]Total time:.*(\d+\.\d+)s"

# Find all matches using the regex pattern
matches = re.findall(pattern, data)

# Organize the extracted information and format the output
results = []
for match in matches:
    benchmark_num, t_value, task_times, total_time = match
    print(match)

    task_times = [int(time) for time in task_times.split(', ')]

    result = {
        'i': int(benchmark_num),
        't': int(t_value),
        'Task Times': task_times,
        'Total Time': float(total_time)
    }
    results.append(result)

# Print the formatted output using JSON
output_json = json.dumps(results, indent=4)
print(output_json)

File: scripts/log-analyzer.js
 
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
  terminal: false
});

let data = '';

// Read data from stdin
rl.on('line', line => {
  data += line + '\n';
}).on('close', () => {
  // Define the regex pattern
  const pattern = /Benchmark (\d+)[^]*?-t (\d+)[^]*?-s (\d+)[^]*?-j (\d+)[^]*?Task times: LongSummaryStatistics\{(.+)\}[^§]*?Total time:.*?(\d+\.\d+)s/g;

  // Find all matches using the regex pattern
  const matches = [...data.matchAll(pattern)];

  // Organize the extracted information and format the output
  const results = [];
  for (const match of matches) {
    let [,i, t, s, j, times, total] = match


    times = parseStatisticsString(times);
    i = parseInt(i);
    t = parseInt(t);
    s = parseInt(s);
    j = parseInt(j);
    total = parseFloat(total);

    const result = {
      i,
      t, s, j,
      ...times,
      total
    };
    results.push(result);
  }

  // Print the formatted output using JSON
  const output_json = JSON.stringify(results, null, 4);
  console.log(output_json);
});

function parseStatisticsString(input) {
  const pattern = /count=(\d+),\s+sum=(\d+),\s+min=(\d+),\s+average=(\d+\.\d+),\s+max=(\d+)/;
  const match = input.match(pattern);

  if (match) {
    return {
      count: parseInt(match[1]),
      sum: parseInt(match[2]),
      min: parseInt(match[3]),
      average: parseFloat(match[4]),
      max: parseInt(match[5])
    };
  } else {
    return null; // Return null for invalid input
  }
}
